### DISCUSSION TITLE
Superset Sorting in bars

### DISCUSSION THREAD

**Discussion by iqbalsingh2988** [apache/superset#29536] [Ideas]
Hi this is my data for the last 1 year containing three columns: Year, Month and Count. When i am creating a bar graph using superset, the bars get arranged alphabetically instead of the order appearing in the result set. How can i do that? Please help.:                                                                               2023	Jul	3997
2023	Aug	6715
2023	Sep	6887
2023	Oct	10730
2023	Nov	9272
2023	Dec	9400
2024	Jan	11004
2024	Feb	10491
2024	Mar	13073
2024	Apr	13706
2024	May	18765
2024	Jun	15474
![image](https://github.com/apache/superset/assets/67512445/b1a170b0-df6b-4599-b039-2c96e56ffc5b)

**Comment by felipegranado** [comment]
Hi!

You are using a column of text to sort. That's why you're not succeeding.

In this case, you need to add a year-month column (ou month number column) to sort correctly (2024-01, 2024-02...).

**Reply by felipegranado** [reply]
Or put the "count" column in "Dimensions" box...

**Comment by sfirke** [comment]
Hello, I don't think this is currently supported (if you want it to go Jan, Feb, Mar etc.) but I have created a more complete feature request here: https://github.com/apache/superset/discussions/30575  I'm going to close this one to consolidate discussions but please reply if that feels inappropriate to you.  I have a link in that discussion to a workaround you can use to achieve this, by padding the front of the character strings with whitespace characters.
